### PR TITLE
gpu: throw error if texture data is not slice

### PR DESCRIPTION
**Pull request by alichraghi**
- [x] By selecting this checkbox, I agree to license my contributions to this project under the license(s) described in the LICENSE file, and I have the right to do so or have received permission to do so by an employer or client I am producing work for whom has this right.

### CODE REVIEW

File: gpu/src/Queue.zig [hunk 1]
```diff
@@ -67,6 +67,9 @@ pub inline fn writeTexture(
     data_layout: *const Texture.DataLayout,
     write_size: *const Extent3D,
 ) void {
+    if (@TypeOf(data) != []const u8) {
```
Comment (slimsag) [inline gpu/src/Queue.zig:70]: Nice! I think there are more places than just here that we need to update - though.

* `Queue.zig` `fn writeBuffer`
* `CommandEncoder.zig` `fn writeBuffer`

Comment (alichraghi) [reply gpu/src/Queue.zig:70]: yeah im aware. will update this soon

Comment (alichraghi) [reply gpu/src/Queue.zig:70]: also this seems wrong (check CI log)

Comment (slimsag) [reply gpu/src/Queue.zig:70]: It looks like `std.meta.isSliceOf(T)` might do what we want https://sourcegraph.com/github.com/ziglang/zig/-/blob/lib/std/meta/trait.zig?L142:7

Comment (alichraghi) [reply gpu/src/Queue.zig:70]: yeah make sense :+1:

**Review by slimsag — APPROVED** [review]
Even better, thank you for doing this!

**Pull request merged**